### PR TITLE
KG-48 Game design without UI

### PR DESCRIPTION
**Pull request by marcelotivelabs**
memory-game code

### CODE REVIEW

**Review by natalyTive — APPROVED** [review]
I dont have more observations

File: __tests__/memoryGame.test.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import MemoryGame from '../src/MemoryGame';
```
Comment (vbrianh) [inline __tests__/memoryGame.test.js:1]: I have the next error:
Screenshot 2023-01-06 at 05 38 44

File: src/MemoryGame.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      this.#clearSelectedCards(selected.selected1, selected.selected2);
+      return { isNewMatched: false }
+    }
+  };
+
+  #verifyWinner () {
+    this.#isThereAWinner = (this.#cards.cardsLength() === this.#cards.getMatches()*2)
+  };
+
+  #verifyCardsSelected (x, y) {
```
Comment (vbrianh) [inline src/MemoryGame.js:108]: remove unnecessary variable x, y and 0

File: src/MemoryGame.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+  #getXLength () {
+    return this.#xLength;
+  };
+
+  #getYLength () {
+    return this.#yLength;
+  };
+
+};
```
Comment (vbrianh) [inline src/MemoryGame.js:143]: remove unnecessary ;

File: src/helpers/inputs.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import readline from 'readline';
+import { stdin as input, stdout as output } from 'node:process';
+
+const rl = readline.createInterface({ input, output });
+
+const ask = (questionText) => {
+  return new Promise((resolve, reject) => {
```
Comment (vbrianh) [inline src/helpers/inputs.js:7]: remove unnecessary variable reject

File: src/utils/validationUtils.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
+import memoryConstants from '../constants/memory.js';
```
Comment (vbrianh) [inline src/utils/validationUtils.js:1]: remove unnecessary variables

**Review by vbrianh — CHANGES_REQUESTED** [review]

File: __tests__/memoryGame.test.js (hunk 1 above)
Comment (marcelotivelabs) [reply __tests__/memoryGame.test.js:1]: It is fixed, please test again

File: src/MemoryGame.js (hunk 2 above)
Comment (marcelotivelabs) [reply src/MemoryGame.js]: done

File: src/MemoryGame.js (hunk 3 above)
Comment (marcelotivelabs) [reply src/MemoryGame.js:138]: done

File: src/helpers/inputs.js (hunk 4 above)
Comment (marcelotivelabs) [reply src/helpers/inputs.js]: done

File: src/utils/validationUtils.js (hunk 5 above)
Comment (marcelotivelabs) [reply src/utils/validationUtils.js:1]: done

File: src/utils/validationUtils.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+const validCategory = (inputCategory) => {
```
Comment (vbrianh) [inline src/utils/validationUtils.js:3]: remove unnecessary ; in line 8, 24, and 32

File: src/MemoryGame.js [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+    return !(counter >= 2 ||
+      isSelected === true ||
+      isMatched === true ||
+      this.#isThereAWinner);
+  };
+
+  #updateRenderCard = (x, y) => {
+    this.#renderCards[x][y] = this.#cards.getCard(`${x}_${y}`).getBack().name;
+  };
+};
```
Comment (vbrianh) [inline src/MemoryGame.js:138]: remove unnecessary ;

File: src/MemoryGame.js [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+      return { isNewMatched: false }
+    }
+  };
+
+  #verifyWinner = () => {
+    this.#isThereAWinner = (this.#cards.cardsLength() === this.#cards.getMatches()*2)
+  };
+
+  #verifyCardsSelected = () => {
+    let counter = null;
```
Comment (vbrianh) [inline src/MemoryGame.js:113]: You can create this variable on the line 116

**Review by vbrianh — CHANGES_REQUESTED** [review]

**Review by vbrianh — APPROVED** [review]
LGTM!

**Pull request merged**